platform/ios/src/MGLMapView.mm

```diff
@@ -656,9 +656,14 @@ - (void)reachabilityChanged:(NSNotification *)notification
     MGLAssertIsMainThread();
 
     MGLReachability *reachability = [notification object];
-    if ( ! _isWaitingForRedundantReachableNotification && [reachability isReachable])
+    if ( ! _isWaitingForRedundantReachableNotification)
     {
-        mbgl::NetworkStatus::Reachable();
+      if ([reachability isReachable]) {
+          mbgl::NetworkStatus::Set(mbgl::NetworkStatus::Status::Online);
+          mbgl::NetworkStatus::Reachable();
+      } else {
+          mbgl::NetworkStatus::Set(mbgl::NetworkStatus::Status::Offline);
+      }
```

Comment:
this was just broken, mapbox wasnt setting network status in the mbgl library, so it was always trying to fetch tiles when offline